src/main/java/de/tum/in/www1/artemis/service/scheduled/cache/quiz/QuizScheduleService.java

```diff
@@ -309,9 +311,9 @@ public void scheduleQuizStart(final long quizExerciseId) {
                     // this is expected if we run on multiple nodes
                 }
             }
+            // Do that at the end because this runs asynchronously and could interfere with the cache write above
+            updateQuizExercise(quizExercise);
         }
-        // Do that at the end because this runs asynchronously and could interfere with the cache write above
-        updateQuizExercise(quizExercise);
```

Comment:
Why was this change needed?

Reply:
good point: the idea was to prevent a potential null pointer exception in case quizExercise is null, but I overlooked the second check in the if statement a couple lines before